### PR TITLE
Added PKeyAuth UserAgent String Support on MacOS

### PR DESCRIPTION
**Pull request by NerevarineRule**
## Proposed changes

Enabled PKeyAuth support via UserAgent String on MacOS
Added an API within MSIDWebviewUIController to return a custom WKWebViewConfig object with PKeyAuth UserAgent String.

## Type of change

- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h [hunk 1]
```diff
@@ -47,6 +48,8 @@ NSWindowController
 @property (nonatomic) UIModalPresentationStyle presentationType;
 #endif
 
++ (WKWebViewConfiguration *)createWebViewConfigWithPKeyAuthUserAgent;
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h:51]: nit: let's call this something more generic? e.g. defaultWKWebViewConfiguration

File: IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m [hunk 2]
```diff
@@ -44,14 +44,26 @@ + (void)initialize
...
-        
+        s_webConfig.applicationNameForUserAgent = kMSIDPKeyAuthKeyWordForUserAgent;
         if (@available(macOS 10.15, *))
         {
             s_webConfig.defaultWebpagePreferences.preferredContentMode = WKContentModeDesktop;
         }
     });
 }
 
++ (WKWebViewConfiguration *)createWebViewConfigWithPKeyAuthUserAgent
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m:55]: can initialize method reuse this API to avoid duplicate code?

**Review by oldalton — CHANGES_REQUESTED** [review]

File: IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m (hunk 2 above)
Comment (NerevarineRule) [reply IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m:55]: Good idea.

File: IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h (hunk 1 above)
Comment (NerevarineRule) [reply IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h:51]: Sure!

File: IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h [hunk 3]
```diff
@@ -47,6 +48,8 @@ NSWindowController
 @property (nonatomic) UIModalPresentationStyle presentationType;
 #endif
 
++ (WKWebViewConfiguration *)defaultWKWebviewConfiguration;
```
Comment (kaisong1990) [inline IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h:51]: Same question from the other CR :)

File: IdentityCore/src/webview/embeddedWebview/ui/ios/MSIDWebviewUIController.m [hunk 4]
```diff
@@ -47,15 +46,22 @@ + (void)initialize
...
+{
+    WKWebViewConfiguration *webConfig = [WKWebViewConfiguration new];
+    webConfig.applicationNameForUserAgent = kMSIDPKeyAuthKeyWordForUserAgent;
+
 #if __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
-        if (@available(iOS 13.0, *))
-        {
-            s_webConfig.defaultWebpagePreferences.preferredContentMode = WKContentModeMobile;
-        }
+    if (@available(iOS 13.0, *))
```
Comment (kaisong1990) [inline IdentityCore/src/webview/embeddedWebview/ui/ios/MSIDWebviewUIController.m:59]: Little confused regarding the logic here, why we have two if checks?

File: IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m (hunk 2 above)
Comment (ameyapat) [reply IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m]: Curious to know, if `defaultWKWebviewConfiguration` is only being called from the singleton in `initialize`, can the method code be shifted to the singleton. Any other class calling this method won't really give the config for this controller.

Comment (NerevarineRule) [reply IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m]: @ameyapat 
Thank you for your comment. 
If we make it private, it won't be accessible in the public ADAL API. I think the usage case would be that OneDrive team could initialize the webview instance and then add the config setting they get from this API using 

WKWebViewConfiguration *webConfig = [MSIDWebviewUIController defaultWKWebviewConfiguration];

WKWebView *webView = [[WKWebView alloc] initWithFrame:rootView.frame configuration:webConfig];

**Review by oldalton — APPROVED** [review]

**Pull request merged**